src/components/ViewMoreText/component.js

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent } from 'react';
+import { Text, View } from 'react-native';
+import { PropTypes } from 'prop-types';
+import { isNull } from 'lodash';
+import { styles } from './styles';
+
+export class ViewMoreText extends PureComponent {
+  state = {
+    showAllText: false,
+    measured: false,
+    fullHeight: null,
+    shouldShowMore: false,
+  }
+
+  onLayout = (event) => {
+    const { fullHeight } = this.state;
+    const height = event.nativeEvent.layout.height;
+    if (isNull(fullHeight)) {
+      this.setState({ fullHeight: height, measured: true });
+      return;
+    }
+
+    if (fullHeight > height) {
+      this.setState({ shouldShowMore: true });
+    }
+  }
+
+  handlePressViewMore = () => {
+    this.setState({ showAllText: true });
+  }
+
+  handlePressViewLess = () => {
+    this.setState({ showAllText: false });
+  }
+
+  renderViewMore = onPress => (
+    <Text style={styles.button} onPress={onPress}>
+      View more
+    </Text>
+  )
+
+  renderViewLess = onPress => (
+    <Text style={styles.button} onPress={onPress}>
+      View less
+    </Text>
+  )
```

Comment:
I've been wondering, is there a reason we use this property style instead of a function syntax?
Since technically this is `function(onPress) {}.bind(this);` but we don't actually reference `this` at all.

Reply:
I'm not entirely sure :/, i've just been trying to use the same standard style across the app. @edencakir may have more answers.

Reply:
I actually would like an answer to this aswell, when do we use a regular function, e.g `abc()` compared to an arrow function `abc = () => {}`. Since i just came to JS i just understand that with the latter you can call `this` directly in it.

I also seem to use the latter syntax more because coming from Swift, any function that is a callback has to be an arrow function, so it might be why i've been writing it like that :/